persistence/schema/gov.go

```diff
@@ -1,442 +1,145 @@
-		"message_unstake_service_node_fee_owner",
-		"message_pause_service_node_fee_owner",
-		"message_unpause_service_node_fee_owner",
-		"message_change_parameter_fee_owner",
-		"end_height",
-	}
+	govParamMetadataMap  map[string]govParamMetadata
+	govParamMetadataKeys []string
+
+	ValTypeEnumTypes = fmt.Sprintf(`(
+		'%s',
+		'%s',
+		'%s'
+		)`,
+		ValTypeString,
+		ValTypeBigInt,
+		ValTypeSmallInt,
+	)
 )
 
 func InsertParams(params *genesis.Params) string {
-	return fmt.Sprintf(`INSERT INTO %s VALUES(%d, '%s', %d, %d, %d, %d, %d, %d,
-						'%s',%d,%d,%d,%d,%d,
-						'%s',%d,%d,%d,%d,
-						'%s',%d,%d,%d,%d,
-						%d,%d,%d,%d,
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',
-						'%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s','%s',%d)`,
-		ParamsTableName,
-		params.BlocksPerSession,
-		params.AppMinimumStake,
-		params.AppMaxChains,
-		params.AppBaselineStakeRate,
-		params.AppStakingAdjustment,
-		params.AppUnstakingBlocks,
-		params.AppMinimumPauseBlocks,
-		params.AppMaxPauseBlocks,
-
-		params.ServiceNodeMinimumStake,
-		params.ServiceNodeMaxChains,
-		params.ServiceNodeUnstakingBlocks,
-		params.ServiceNodeMinimumPauseBlocks,
-		params.ServiceNodeMaxPauseBlocks,
-		params.ServiceNodesPerSession,
-
-		params.FishermanMinimumStake,
-		params.FishermanMaxChains,
-		params.FishermanUnstakingBlocks,
-		params.FishermanMinimumPauseBlocks,
-		params.FishermanMaxPauseBlocks,
-
-		params.ValidatorMinimumStake,
-		params.ValidatorUnstakingBlocks,
-		params.ValidatorMinimumPauseBlocks,
-		params.ValidatorMaxPauseBlocks,
-		params.ValidatorMaximumMissedBlocks,
+	val := reflect.ValueOf(params)
+	var subQuery string
+	for _, k := range govParamMetadataKeys {
+		pnt := govParamMetadataMap[k]
+		pVal := val.Elem().FieldByName(pnt.PropertyName)
+
+		subQuery += `(`
+		switch govParamMetadataMap[k].PoktValType {
+		case ValTypeString:
+			subQuery += fmt.Sprintf("'%s', %d, true, '%s', '%s'", k, DefaultBigInt, pnt.PoktValType, pVal.Interface())
+		case ValTypeSmallInt, ValTypeBigInt:
+			subQuery += fmt.Sprintf("'%s', %d, true, '%s', %d", k, DefaultBigInt, pnt.PoktValType, pVal.Interface())
+		}
+		subQuery += `),`
+	}
+	return fmt.Sprintf(`INSERT INTO %s VALUES %s`, ParamsTableName, subQuery[:len(subQuery)-1])
+}
 
-		params.ValidatorMaxEvidenceAgeInBlocks,
-		params.ProposerPercentageOfFees,
-		params.MissedBlocksBurnPercentage,
-		params.DoubleSignBurnPercentage,
+func GetParamQuery(paramName string) string {
+	//TODO (@deblasis): clarify if and how we should use `enabled` here
+	return fmt.Sprintf(`SELECT value FROM %s WHERE name=%s AND height=%d and enabled=true`, ParamsTableName, paramName, DefaultBigInt)
```

Comment:
I think I need to better understand your train of thought around the `enabled` flag. Not sure if this is correct,

Reply:
Re height - I think we'll also need to do a `where height<=%d AND enabled=true order by height desc limit 1`.

CC @andrewnguyen22  for confirmation.